src/librustdoc/json/mod.rs

```diff
@@ -183,21 +183,8 @@ impl<'tcx> FormatRenderer<'tcx> for JsonRenderer<'tcx> {
         Ok(())
     }
 
-    fn mod_item_in(&mut self, item: &clean::Item, _module_name: &str) -> Result<(), Error> {
-        use clean::types::ItemKind::*;
-        if let ModuleItem(m) = &*item.kind {
-            for item in &m.items {
-                match &*item.kind {
-                    // These don't have names so they don't get added to the output by default
-                    ImportItem(_) => self.item(item.clone()).unwrap(),
-                    ExternCrateItem { .. } => self.item(item.clone()).unwrap(),
-                    ImplItem(i) => i.items.iter().for_each(|i| self.item(i.clone()).unwrap()),
-                    _ => {}
-                }
-            }
-        }
-        self.item(item.clone()).unwrap();
-        Ok(())
+    fn mod_item_in(&mut self, _item: &clean::Item, _module_name: &str) -> Result<(), Error> {
+        unreachable!()
     }
 
     fn mod_item_out(&mut self, _item_name: &str) -> Result<(), Error> {
```

Comment:
nit: This can also be make `unreachable!`

Reply:
Oh hmm, I think I was assuming this already had the changes from https://github.com/rust-lang/rust/pull/84458/files#diff-46f011cf734f3de35eddca150990366cc41bd8eaf3945a241b2a2f79c5bb77eaR40. Do you think it's important to add the assert given that this function gets removed altogether there? Or is it fine to have it silently do nothing? If the latter I'd prefer to leave this as-is to avoid merge conflicts.

Reply:
I think its fine to leave then, as it would be very wierd to call `mod_item_out` without `mod_item_in`